src/components/FormList.js

```diff
@@ -0,0 +1,131 @@
+
+
+import React, {useState} from 'react';
+import {getTodaysDate} from "./Date";
+
+// import { useForm } from "react-hook-form";
+
+const FormList = (props) => {
+  // const todays = new Date();
+  // console.log(todays);
+  const [title,setTitle] = useState("");
+  const [sdate,setSdate] = useState(getTodaysDate());
+  const [edate,setEdate] = useState(getTodaysDate());
+  const [atleast,setAtleast] = useState("1");
+  const [singular,setSingular] = useState("");
+  const [plural,setPlural] = useState("");
+  const oneDay = 24*60*60*1000;
+  const starDate = new Date(sdate);
+  const endDate = new Date(edate); 
+  const differenceDates = Math.round(Math.abs((starDate - endDate)/oneDay));
+  console.log(differenceDates);
+
+  
+  // var opVal
+  // for(var i=0; i<= 8; i++){
+    //   opVal = periodSelection[i].value;
+    // }
+    // console.log(opVal);
+    const submitHandler = (e)=>{
+      var periodSelection = document.getElementById("datePeriod").getElementsByTagName("option");
+    e.preventDefault();
+    // console.log(title);
+    // console.log(sdate);
+    // console.log(edate);
+    // console.log(atleast);
+    // console.log(singular);
+    // console.log(plural);
+  if(title===""){
+    document.getElementById("title").innerText="Title should not be blank";
+  }
+  else if(edate<sdate){
+    document.getElementById("edate").innerText="End date should not be less than current date";
+  }
+  else if(atleast===""){
+    document.getElementById("atleast").innerText="Atleast should not be blank";
+    
+    
+  }
+  else if(singular===""){
+    document.getElementById("title").innerText="";
+    document.getElementById("singular").innerText="Singular should not be blank";
+  }
+  else if(plural===""){
```

Comment:
The conditional logic looks solid here, but instead of setting validation messages with `innerHTML`, we should just rely on React rendering these messages conditionally.

I guess it would probably make sense to extend the `InputField` and the `Select` components with a validation error prop, just to make this easy going forward.